### PR TITLE
feat(banner): moved banner logic from main package to add-on package

### PR DESCRIPTION
**Pull request by rsebade**
Created a new PR since the code is so different!

### CODE REVIEW

File: example/src/App.js [hunk 1]
```diff
@@ -1,10 +1,377 @@
...
+            <Content />
+        </DotcomShell>
+      </>
+      )
 }
 
 export default App
+
+
+// {
```
Comment (annawen1) [inline example/src/App.js]: we don't need this commented out section right?

File: README.md [hunk 2]
```diff
@@ -15,22 +15,46 @@ As recommended by create-react-library docs, it’s better to open two terminal
...
 ```
 
+Also, create a `.env` file in the example directory and add the following to ensure the sass path compiles correctly
+```
+SASS_PATH=node_modules:src
+```
+
 ## Usage
 
+The most common usage will be to replace the Masthead/DotcomShell component from @carbon/ibmdotcom-react with the component in this package. These components have the same props as the original, except they also have a `hasBanner` prop. If `hasBanner` is enabled, the think-banner will be displayed above the masthead.
```
Comment (annawen1) [inline README.md]: can you also remove this old readme usage section?

File: example/src/App.js (hunk 1 above)
Comment (rsebade) [reply example/src/App.js]: Nope, that was just a test object for an example L1, will remove!

File: README.md (hunk 2 above)
Comment (rsebade) [reply README.md]: Yep!

File: src/styles.scss [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+  #masthead-banner-mobile {
+    display: none;
+  }
+}
+
+// THINK BANNER 2020
+#think-banner-link {
+  text-decoration: none;
+  background: #f4f5f6;
+  color: #000000;
```
Comment (ariellalgilmore) [inline src/styles.scss:98]: Wondering if we can use carbon tokens? (`$ui-background`)

**Review by ariellalgilmore — COMMENTED** [review]
noticing some weird things happening to the masthead...
https://user-images.githubusercontent.com/20210594/116944626-b62df080-ac2a-11eb-957b-eaabbafaba13.mov

**Comment by rsebade** [comment]
> noticing some weird things happening to the masthead...
> https://user-images.githubusercontent.com/20210594/116944626-b62df080-ac2a-11eb-957b-eaabbafaba13.mov

Not sure why the search icon is off center in the button... looks like clicking it probably rerenders the component & removes the class, I'll probably extend the mutation observer I had set up to add the class back when this happens. As for the login menu, I'll have to add a class to it as well to adjust it's positioning.

**Comment by ariellalgilmore** [comment]
> > noticing some weird things happening to the masthead...
> > https://user-images.githubusercontent.com/20210594/116944626-b62df080-ac2a-11eb-957b-eaabbafaba13.mov
> 
> Not sure why the search icon is off center in the button... looks like clicking it probably rerenders the component & removes the class, I'll probably extend the mutation observer I had set up to add the class back when this happens. As for the login menu, I'll have to add a class to it as well to adjust it's positioning.

okay makes sense! Thanks @rsebade! The class for the login menu should be applied for the collapsed hamburger menu too

File: src/styles.scss (hunk 3 above)
Comment (rsebade) [reply src/styles.scss:98]: The reason I have it set up the way it is is to keep it as aligned with the v18 version as possible, which means sharing all styling. Keeping that in mind, I think it's safer to match the css exactly where possible, at least until we fully deprecate the v18 version.

**Comment by rsebade** [comment]
> > > noticing some weird things happening to the masthead...
> > > https://user-images.githubusercontent.com/20210594/116944626-b62df080-ac2a-11eb-957b-eaabbafaba13.mov
> > 
> > 
> > Not sure why the search icon is off center in the button... looks like clicking it probably rerenders the component & removes the class, I'll probably extend the mutation observer I had set up to add the class back when this happens. As for the login menu, I'll have to add a class to it as well to adjust it's positioning.
> 
> okay makes sense! Thanks @rsebade! The class for the login menu should be applied for the collapsed hamburger menu too

@ariellalgilmore the latest commit should fix everything in that video, let me know if you're still seeing it.

File: src/styles.scss (hunk 3 above)
Comment (annawen1) [reply src/styles.scss:98]: that makes sense, thanks for explaining!

**Comment by annawen1** [comment]
@rsebade looks like we need to set the `top` property to `48px` when scrolling along the page and then opening the megamenu

Screen Shot 2021-05-04 at 9 32 29 AM

**Comment by ariellalgilmore** [comment]
> @rsebade looks like we need to set the `top` property calculation when scrolling along the page and then opening the megamenu
> 
> Screen Shot 2021-05-04 at 9 32 29 AM

looks a lot better @rsebade! yes relating to the megamenu `top`, I see the same problem happening with the profile menu drop down when the user scrolls.

**Comment by rsebade** [comment]
> @rsebade looks like we need to set the `top` property to `48px` when scrolling along the page and then opening the megamenu
> 
> Screen Shot 2021-05-04 at 9 32 29 AM

Just pushed a fix! Yeah since the megamenus are position:fixed I had to adjust their top position on scroll when the banner was in the viewport. It seems like the scroll handler wasn't firing fast enough though when the user got past the banner and the top was left too high. I gave it a threshold of an extra 50 pixels to change the top position if necessary and it seems to be working now.

**Review by annawen1 — APPROVED** [review]
LGTM! Thanks Ryan!

**Pull request merged**